src/nspawn/nspawn-settings.c

```diff
@@ -1019,3 +1021,57 @@ int config_parse_bind_user(
 
         return 0;
 }
+
+int parse_bind_user_shell(const char *s, char **ret) {
+        int r;
+        char *sh;
+
+        if (is_path(s))
+                sh = strdup(s);
+        else {
+                r = parse_boolean(s);
+                if (r < 0)
+                        return r;
+
+                sh = strdup(r ? "yes" : "no");
```

Comment:
I do not think it is necessary to copy "no". Please set NULL.

Reply:
It's important to distinguish between `BindUserShell=no` and the lack of the setting due to the way settings and CLI options interact [here](https://github.com/systemd/systemd/pull/37921/files#diff-3e88b6405bcaf989c1ff90c722b131c15b11bc1c3856c2b1a6fb375f12efafa5R4860).

I think we want the settings to override the CLI in the case of `--settings=override --bind-user-shell=yes` with an explicit `BindUserShell=no`, but we don't want it to override if the setting is not present or empty.

Reply:
> It's important to distinguish between `BindUserShell=no` and the lack of the setting due to the way settings and CLI options interact [here](https://github.com/systemd/systemd/pull/37921/files#diff-3e88b6405bcaf989c1ff90c722b131c15b11bc1c3856c2b1a6fb375f12efafa5R4860).
> 
> I think we want the settings to override the CLI in the case of `--settings=override --bind-user-shell=yes` with an explicit `BindUserShell=no`, but we don't want it to override if the setting is not present or empty.

Why not to simply drop the condition you pointed?? Do I misunderstand or overlooked something?

Reply:
Gotcha, so rather than a single char* in one of the four non-overlapping states (`settings->bind_user_shell \in {"/.*", "yes", "no", NULL}`) we'll need three distinct settings:
- `settings->bind_user_shell \in {path, NULL}` (an absolute path, if specified, or implicit/explicit `=no` when `NULL`),
- `settings->bind_user_shell_copy \in {0, 1}` (indicate `=yes` state, required in `convert_user`), and
- `settings->bind_user_shell_set \in {0, 1}` (similar to `oom_score_adjust_set`, required to indicate when the `=no` state indicated by `settings->bind_user_shell = NULL` must override an explicit `--bind-user-shell=` CLI option in `merge_settings`).

Glad to make the change, if this structure is preferred.